toolchain/check/pattern_match.cpp

```diff
@@ -141,12 +141,9 @@ auto MatchContext::EmitPatternMatch(Context& context,
     case SemIR::BindingPattern::Kind:
     case SemIR::SymbolicBindingPattern::Kind: {
       CARBON_CHECK(kind_ == MatchKind::Callee);
-      auto binding_pattern = pattern.inst.As<SemIR::AnyBindingPattern>();
-      auto cache_entry =
-          context.bind_name_cache().Lookup(binding_pattern.entity_name_id);
-      // The cached bind_name should only be used once.
-      auto bind_name_id =
-          std::exchange(cache_entry.value(), SemIR::InstId::Invalid);
+      auto [bind_name_id, type_expr_id] =
+          context.bind_name_map().Lookup(entry.pattern_id).value();
```

Comment:
Would it be incorrect to remove the pattern id from the map after this? Is it used more than once? The InsertHere() below feels like that must mean it isn't but I am not confident.